### DISCUSSION TITLE
有人说如果部分开源（把解密的代码抠掉）要有协议，对此龙猫你怎么看

### DISCUSSION THREAD

**Discussion by Lyrambra** [Meloong-Git/PCL#2513] [提问] [answered]
![image](https://github.com/Hex-Dragon/PCL2/assets/120571796/79b9b354-5c70-4e9c-bc6b-201ecf8d6928)
![image](https://github.com/Hex-Dragon/PCL2/assets/120571796/8134157c-922a-4bb8-bd08-93da22249eb8)

**Comment by hejiehao** [comment]
有很多软件都是这样子的，比如Jetbrains全家桶，社区版是开源的，但是专业版是闭源的还收费

**Reply by Lyrambra** [reply]
你或许没理解我的意思，我指的是龙猫在公测版的源代码里把隐藏主题和主页的解谜游戏源码故意扣掉
![f45683b7d26ea6e2b86aec84834539a5](https://github.com/Hex-Dragon/PCL2/assets/120571796/5c1e4592-f742-4d0e-90d5-1f01644237b4)

**Reply by hejiehao** [reply]
开源不是免费

**Comment by LTCatt** [comment]
就像楼上说的，我没听过哪里有截图那位的这种说法

**Reply by Lyrambra** [reply]
好的，明白了，我会转达

**Reply by Lyrambra** [reply]
等等，对方又给出了新的观点

**Reply by Lyrambra** [reply]
![_NS{)V%KO@KYXZWN147}TTD_tmb](https://github.com/Hex-Dragon/PCL2/assets/120571796/a0f10f62-133d-4fc3-a6db-a0a0b46b04d6)
![)$P PNWC(UK29 MQ}W)DM`A_tmb](https://github.com/Hex-Dragon/PCL2/assets/120571796/162423a9-565d-4c2d-b7c0-194a815ae5c3)

**Reply by Lyrambra** [reply]
![image](https://github.com/Hex-Dragon/PCL2/assets/120571796/56add330-52dc-4af0-94c5-09374f80852b)
![image](https://github.com/Hex-Dragon/PCL2/assets/120571796/81b9c261-4da8-402c-ae28-7deb07a34935)

**Reply by Lyrambra** [reply]
而且AI的回答是这样的

**Reply by Lyrambra** [reply]
对于一个开源软件平台如GitHub上的开源项目，如果其中的某个应用的彩蛋代码被作者故意隐藏且没有提供相关获取渠道，这种情况下有以下几种可能性：

违反开源许可协议：如果该开源项目采用了某个开源许可协议（如GPL、MIT等），则作者隐藏彩蛋代码并未提供获取渠道可能违反了该许可协议的规定。大部分开源许可协议要求全部源代码都必须公开和可获得。

不符合开源软件的原则：即使没有使用特定的开源许可协议，隐藏彩蛋代码并且没有相关获取渠道的行为也可能被视为不符合开源软件的基本原则。开源软件的核心概念是透明和开放，隐藏一部分代码并无法获得可能被视为不尊重开源原则。

需要与作者沟通：作为用户，如果您对该开源项目中的彩蛋代码感兴趣，可以尝试与作者进行沟通，了解他们对于隐藏代码和获取渠道的立场。或许是作者有特定的原因选择隐藏彩蛋代码，但如果这违背了开源原则，可能需要进行讨论和解决。

总之，对于位于开源软件平台上的开源项目，隐藏彩蛋代码并未提供获取渠道可能违反开源许可协议或开源原则。但最终的结论还需要根据具体的情况和开源许可协议来确定，同时与作者进行沟通和讨论可能是解决问题的一种方式。

**Reply by Lyrambra** [reply]
![image](https://github.com/Hex-Dragon/PCL2/assets/120571796/3cb25619-ee2b-4923-88b3-c34b11379399)

**Reply by Lyrambra** [reply, marked as answer]
等等，我明白了，AI帮我大忙
![image](https://github.com/Hex-Dragon/PCL2/assets/120571796/ed175f8c-011b-47fe-8509-75fc664c758e)

**Reply by Chiloven945** [reply]
是这样，github本身就是一个代码分享的，是分享不是在这交税=。=
而且龙猫应该也没有说过使用了什么开源协议，所以说最终的解释还是在版权所有者手里，扯这些纯粹没事找事

**Comment by Chiloven945** [comment]
版权拥有者保留所有权利，解释权也通通归版权所有者。所以说根本不需要写用什么协议直接license里写All Rights Reserved就行了，看到侵权行为直接警告。开源又不是需要全部给开源=。=

**Comment by so1ve** [comment]
这人是玩GitHub都没玩懂吗没有协议的话不是作者保留解释权么哪轮得到这么个跳梁小丑出来取乐大家 😅 开源又不是免费 那么多项目都是挂个readme结果没有源代码怎么没见他去冲 真是让他丑完了
